File: ProgAndPlay/Client_Interfaces/python/mission4.py
from constantList_KP4_1 import *
from pp import *

# définition des positions
bitsPos = PP_Pos(1400.0, 1371.0)
bytesPos = PP_Pos(479.0, 1825.0)
# Ouverture de la connexion avec le jeu
PP_Open()
# Récupérer la première unité
u = PP_GetUnitAt(MY_COALITION, 0)
if PP_Unit_GetType(u) == BIT:
    # Ordonner à mon unité courante de se déplacer
    PP_Unit_ActionOnPosition(u, MOVE, bitsPos)
    # Passer à l'unité suivante
    u = PP_GetUnitAt(MY_COALITION, 1)
    # Lui ordonner de rejoindre sa position
    PP_Unit_ActionOnPosition(u, MOVE, bytesPos)
else:
    # Ordonner à mon unité courante de se déplacer
    PP_Unit_ActionOnPosition(u, MOVE, bytesPos)
    # Passer à l'unité suivante
    u = PP_GetUnitAt(MY_COALITION, 1)
    # Lui ordonner de rejoindre sa position
    PP_Unit_ActionOnPosition(u, MOVE, bitsPos)
# Fermer la connexion avec le jeu
PP_Close()

File: ProgAndPlay/Client_Interfaces/c-cpp/exemple/mission4.c
#include "PP_Client.h"
#include "../constantList_KP4.1.h"

// déplacer le bit et le byte à leurs coordonnées respectives
int main (){
	// définition des positions
	PP_Pos bits;
	bits.x = 1400.0;
	bits.y = 1371.0;
	PP_Pos bytes;
	bytes.x = 479.0;
	bytes.y = 1825.0;
	// ouverture du jeu
	PP_Open();
	// récupérer la première unité
	PP_Unit u = PP_GetUnitAt(MY_COALITION, 0);
	if (PP_Unit_GetType(u) == BIT){
		// ordonner à mon unité courante de se déplacer
		PP_Unit_ActionOnPosition(u, MOVE, bits);
		// passer à l'unité suivante
		u = PP_GetUnitAt(MY_COALITION, 1);
		// lui ordonner de rejoindre sa position
		PP_Unit_ActionOnPosition(u, MOVE, bytes);
	}
	else {
		// ordonner à mon unité courante de se déplacer
		PP_Unit_ActionOnPosition(u, MOVE, bytes);
		// passer à l'unité suivante
		u = PP_GetUnitAt(MY_COALITION, 1);
		// lui ordonner de rejoindre sa position
		PP_Unit_ActionOnPosition(u, MOVE, bits);
	}
	// fermer le jeu
	PP_Close();
}
